### PR TITLE
fix: pre-commit ci workflow

### PR DESCRIPTION
**Pull request by kostasrim**
* fix pip install to run under venv
* fix fakeredis python installation

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,6 +26,17 @@ jobs:
           lsblk -l
           echo "sda rotational = $(cat /sys/block/sda/queue/rotational)"
           echo "sdb rotational = $(cat /sys/block/sdb/queue/rotational)"
+      - name: Create virtual environment
```
Comment (adiholden) [inline .github/workflows/ci.yml:29]: I meant that you need to change the current Install dependencies step line 22 to run in virtual env

Comment (kostasrim) [reply .github/workflows/ci.yml:29]: yep yep, I was distracted with the gh runner :stuck_out_tongue:

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,17 +21,26 @@ jobs:
       - uses: actions/setup-python@v5
       - name: Install dependencies
         run: |
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install --upgrade pip
```
Comment (adiholden) [inline .github/workflows/ci.yml:26]: why?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,17 +21,26 @@ jobs:
...
           python -m pip install pre-commit
           python -m pip freeze --local
           lsblk -l
           echo "sda rotational = $(cat /sys/block/sda/queue/rotational)"
           echo "sdb rotational = $(cat /sys/block/sdb/queue/rotational)"
+      - name: Install dependencies
+        run: |
+          source venv/bin/activate
+          pip install pre-commit
```
Comment (adiholden) [inline .github/workflows/ci.yml:35]: why again?

Comment (kostasrim) [reply .github/workflows/ci.yml:35]: because among different `run` commands you need to `source` again

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kostasrim) [reply .github/workflows/ci.yml:26]: The > python: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found  
is not needed. Will remove

File: .github/workflows/ci.yml (hunk 3 above)
Comment (kostasrim) [reply .github/workflows/ci.yml:35]: (I am testing this not 100% sure about this)

File: .github/workflows/test-fakeredis.yml [hunk 4]
```diff
@@ -39,18 +36,18 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: true
-      - uses: actions/setup-python@v5
```
Comment (kostasrim) [inline .github/workflows/test-fakeredis.yml:42]: For some reason this caused a broken installation of python. I fell back to the one provided by ubuntu (3.10 for 22.04)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -21,8 +21,9 @@ jobs:
       - uses: actions/setup-python@v5
       - name: Install dependencies
         run: |
-          python -m pip install pre-commit
-          python -m pip freeze --local
+          python -m venv venv
+          source venv/bin/activate
+          pip install pre-commit
```
Comment (adiholden) [inline .github/workflows/ci.yml:26]: why not using python -m?
why did you remove `python -m pip freeze --local`

File: .github/workflows/test-fakeredis.yml (hunk 4 above)
Comment (kostasrim) [reply .github/workflows/test-fakeredis.yml:42]: Yes I think the `cache-dependency-path` was the problem but I did not try removing it and keeping the `actions/setup`. We want the the python version that comes with the system and the best way to do that is `apt install` and not relying on some action to fetch the dependency

File: .github/workflows/ci.yml (hunk 5 above)
Comment (kostasrim) [reply .github/workflows/ci.yml:26]: 1. Does it really matter if I use `python -m` or `pip install` directly ?
2. Because it's irrelevant -- it freezes the current state of the virtual env such that it can be recreated somewhere else. We dont't need that here

**Comment by romange** [comment]
This https://github.com/dragonflydb/dragonfly/pull/3924 fixes the problem. Really no need for all this virtual env stuff in our CI

**Comment by kostasrim** [comment]
> This #3924 fixes the problem. Really no need for all this virtual env stuff in our CI

---break-system-packages

Can break the system
